src/components/wizards/create-vm/CreateVMWizard.js

```diff
@@ -0,0 +1,366 @@
+import React from 'react';
+import { Form, FormControl, Col, ButtonGroup, DropdownButton, MenuItem, Checkbox, FormGroup, ControlLabel, HelpBlock, Wizard } from 'patternfly-react';
+import './CreateVMWizard.css';
+import * as _ from 'lodash';
+import { StorageTable } from './StorageTable';
+
+const defaultState = {
+  activeStepIndex: 0,
+  basicSettings: {},
+  wizardValid: false
+}
+
+export class CreateVMWizard extends React.Component {
+
+    constructor(props){
+      super(props);
+      this.state = defaultState;
+    }
+  
+    onFormChange = (newValue, target) => {
+      let valid = null;
+      if (this.basicSettings[target].required && newValue.trim().length === 0){
```

Comment:
`this.basicSettings` represents fields of form (type, label, validation). `this.state.basicSettings` represents actual values that are changing based on user input (value, validation message). Maybe I should rename one of them to avoid confusion

Reply:
I see, I'd rename them to better reflect their purpose, i.e. `this.basicStepFields` and `this.state.basicStepValues` or similar.

I'd use the word "step" since it's the grouping concept that puts multiple form fields together, from visual/user perspective.